lib/jsonrpc.ml

```diff
@@ -554,27 +554,41 @@ let get name dict =
     raise (Malformed_method_request str)
   | Some v -> v
 
-let call_of_string str =
+let version_and_call_of_string str =
   match of_string str with
   | Dict d ->
-    let name = match get "method" d with String s -> s | _ -> raise (Malformed_method_request str) in
-    let params =
+    let name =
+      match get "method" d with
+      | String s -> s
+      | _ -> raise (Malformed_method_request "method")
```

Comment:
`call_of_rpc` was renamed to `version_and_call_of_rpc`. And then a "new" `call_of_rpc` was added for compat, which calls `version_and_call_of_rpc`. So you can catch a new exception from `version_and_call_of_rpc` in `call_of_rpc` and reraise as the old one.